File: Python/UmActually/umActually.py
import re

def um_actually(phrase):
  lowercase_phrase = phrase.lower()
  final_phrase = ""
  character_count = 0

  for i in lowercase_phrase:
    if (re.search("[^a-zA-Z0-9]", i)):
      final_phrase += i
      character_count+= 1
    else:
      if (character_count % 3 == 0):
        final_phrase += i.upper()
        character_count+= 1
      else:
        final_phrase += i
        character_count+= 1
    

  print(final_phrase)

um_actually("Inertia is a property of matter")

File: C++/UmActually/UmActually.cpp
#include <iostream>
#include <regex>

std::string umActually(std::string phrase) {
    std::transform(phrase.begin(), phrase.end(), phrase.begin(), ::tolower);
    std::string finalPhrase;
    std::regex pattern("[^a-zA-Z]");

    for (int i = 0; i < phrase.length(); i++) {
        if (std::regex_match(phrase, pattern)) {
            finalPhrase += phrase[i];
        } else {
            if (i % 3 == 0) {
                finalPhrase += std::toupper(phrase[i]);
            } else {
                finalPhrase += phrase[i];
            }
        }
    }

    return finalPhrase;
}

int main() {
    std::cout << umActually("Inertia is a property of matter") << std::endl;
    return 0;
}
